src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -65,7 +75,8 @@ public SchemaObjectType(ObjectSchema objectSchema, BuildContext context)
         protected override TypeKind TypeKind => IsStruct ? TypeKind.Struct : TypeKind.Class;
         public bool IsStruct => ExistingType?.IsValueType == true;
 
-        public ObjectSerialization[] Serializations => _serializations ??= BuildSerializations();
+        public JsonObjectSerialization? JsonSerialization => _hasJsonSerialization ? _jsonSerialization ??= _serializationBuilder.BuildJsonObjectSerialization(ObjectSchema, this) : null;
```

Comment:
Collection of serializations has been replaced with 2 explicit fields. This has made logic in SerializationWriter more straightforward.